File: RachitGoyal/Bubblesort.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    int n;
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++)
    {
        cin>>arr[i];
    }
    int temp;
    for(int i=0;i<n-1;i++)
    {
        for(int j=0;j<n-i-1;j++)
        {
            if(arr[j+1]<arr[j])
            {
                temp=arr[j];
                arr[j]=arr[j+1];
                arr[j+1]=temp;
            }
        }
        
    }
    for(int i=0;i<n;i++)
    {
        cout<<arr[i]<<' ';
    }
     
     return 0;
}


//Space Complexity: O(1)
//Time Complexity:-
//Worst Case:O(n^2)
//Best Case: O(n)

File: Prashant/Sorting/BubbleSort.java

/** this program shows the implementation of bubble sort */
import java.io.IOException;
import java.util.Scanner;

public class BubbleSort {
    public static void main(String[] args) {
        int[] a = { 10, 9, 7, 101, 23, 44, 12, 78, 34, 23 };
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                if (a[i] < a[j]) {
                    int temp = a[i];
                    a[i] = a[j];
                    a[j] = temp;
                }
            }
        }
        System.out.println("Printing Sorted List ...");
        for (int i = 0; i < 10; i++) {
            System.out.println(a[i]);
        }
    }
}

// Worst Case Time Complexity [ Big-O ]: O(n2)
// Best Case Time Complexity [Big-omega]: O(n)
// Average Time Complexity [Big-theta]: O(n2)
// Space Complexity: O(1)
